app.rb

```diff
@@ -0,0 +1,177 @@
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase == 'y'
+
+    student = Student.new(name, age, parent_permission: parent_permission)
+    @people << student
+    puts 'Student created successfully'
+  end
+
+  def create_teacher
+    puts 'Create a Teacher'
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+
+    teacher = Teacher.new(name, age, parent_permission: true, specialization: specialization)
+    @people << teacher
+    puts 'Teacher created successfully'
+  end
+
+  def create_book
+    puts 'Create a Book'
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def book_number
+    puts 'Select a book from the following list by number:'
+    display_books
+    gets.chomp.to_i
+  end
+
+  def person_number
+    puts 'Select a person from the following list by number:'
+    display_people
+    gets.chomp.to_i
+  end
+
+  def create_rental
+    book_number = book_number_from_user
+    person_number = person_number_from_user
+    puts 'Enter rental date (YYYY-MM-DD):'
+    date = gets.chomp
+
+    rental = Rental.new(date, @books[book_number], @people[person_number])
+    @rentals << rental
+    puts 'Rental created successfully'
+  end
+
+  def book_number_from_user
+    puts 'Select a book from the following list by number:'
+    display_books
+    gets.chomp.to_i
+  end
+
+  def person_number_from_user
+    puts 'Select a person from the following list by number:'
+    display_people
+    gets.chomp.to_i
+  end
+
+  def display_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def display_people
+    @people.each_with_index do |person, index|
+      if person.is_a?(Teacher)
+        puts "#{index}) [Teacher] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      elsif person.is_a?(Student)
+        puts "#{index}) [Student] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      else
+        puts "#{index}) Person Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def list_rentals_for_person
+    puts 'Select a person from the following list by number:'
+    display_people
+    person_number = gets.chomp.to_i
+
+    person = @people[person_number]
+    unless person
+      puts 'Invalid person number'
+      return
+    end
+
+    rentals = @rentals.select { |rental| rental.person == person }
+    if rentals.empty?
+      puts "No rentals found for #{person.name}"
+    else
+      puts "Rentals for #{person.name}:"
+      rentals.each { |rental| puts "#{rental.book.title}, Date: #{rental.date}" }
+    end
```

Comment:
- [ Optional ] I think it would be better if you list the rentals a person has using the person's `id`, according to the project requirements the rentals of a person should be listed using the `id` of the person and not the position of the person on the user's list.

## Project requirements
![Screenshot from 2023-06-23 12-19-00](https://github.com/naji-97/OOP-school-library/assets/16392046/eb6217a2-be29-43de-86e2-f5327e34f066)